gammapy/visualization/utils.py

```diff
@@ -280,3 +286,111 @@ def plot_theta_squared_table(table):
+    """
+    Plot the 1D distribution of data inside a map as a histogram. If the dimension of the map is smaller than 2,
+    a unique plot will be displayed. Otherwise, if the dimension is 3 or greater, a grid of plot will be displayed.
+
+    Parameters
+    ----------
+    map_ : an instance of `~gammapy.maps.Map`
+        A map that contains data to be plotted.
+    ax : `~matplotlib.axes.Axes` or list of `~matplotlib.axes.Axes`
+        Axis object to plot on. If a list of Axis is provided it has to be the same length as the length of _map.data.
+    ncols : int
+        Number of columns to plot if a "plot grid" was to be done.
+    fit : bool
+        Whether to perform a fit of the distribution of data. If True, the fit is performed by `scipy.stats`.
+    fit_function : str
+        A string of a name of a scipy continuous distribution from `scipy.stats._continuous_distns`.
+    **kwargs : dict
+        Keyword arguments to pass to `matplotlib.pyplot.hist` and `matplotlib.axes.Axes`.
+
+    Returns
+    -------
+    fit_parameters : list of tuple
+        List of tuple of the fitted parameters corresponding to the chosen `fit_function`. If `fit` is set to False,
+        returns an empty list.
+    axes : `~numpy.ndarray` of `~matplotlib.pyplot.Axes`
+        Array of Axes.
+    """
+
+    if not isinstance(map_, Map):
+        raise TypeError(
+            f"map_ must be an instance of gammapy.maps.Map, given {type(map_)}"
+        )
+
+    hist_args = list(inspect.signature(plt.hist).parameters)
+    patches_args = list(inspect.signature(matplotlib.patches.Patch).parameters)
+    all_hist_args = np.unique(hist_args + patches_args)
+
+    hist_dict = {k: kwargs.pop(k) for k in dict(kwargs) if k in all_hist_args}
+
+    axes_dict = {k: kwargs.pop(k) for k in dict(kwargs) if k not in hist_args}
+
+    data = map_.data
+    if data.ndim == 1:
+        data = np.reshape(
+            data,
+            (
+                1,
+                1,
+            )
+            + data.shape,
+        )
+    elif data.ndim == 2:
+        data = np.reshape(data, (1,) + data.shape)
+
+    if ax is None:
+        n_plot = len(data)
+        cols = min(ncols, n_plot)
+        rows = 1 + (n_plot - 1) // cols
+
+        width = 12
+        figsize = (width, width * rows / cols)
+
+        fig, axes = plt.subplots(
+            nrows=rows,
+            ncols=cols,
+            figsize=figsize,
+        )
+        cells_in_grid = rows * cols
+    else:
+        axes = ax
+        cells_in_grid = len(ax.flatten())
+
+    if not isinstance(axes, np.ndarray):
+        axes = np.array([axes])
+
+    for idx in range(cells_in_grid):
+
+        axe = axes.flat[idx]
+        if idx > len(data) - 1:
+            axe.set_visible(False)
+            continue
+        d = data[idx]
+        axe.hist(d.flatten(), **hist_dict)
+
+        fit_parameters = []
+        if fit:
+
+            function = getattr(scipy.stats._continuous_distns, fit_function)
```

Comment:
We should not rely on private namespaces like `_continuous_distns`, probably the distributions are just visible in `scipy.stats` as well? If not users can also just pass the method themselves (so not via a string but just the actual function object).

Reply:
The thing is that there are not continuous distribution in `scipy.stats` , and I am not sure that we would want that. 
But I guess it's fine if the user pass directly the object.